### ISSUE TITLE
Is require-concurrent-index-creation necessary straight after table creation?

### ISSUE THREAD

**Issue by maciej** [sbdchd/squawk#349]
Currently if even if an index is created directly after the table it refers to `require-concurrent-index-creation` will be triggered. Index creation on an empty table is virtually instantaneous and locking a table that is unlikely to receive writes until the migration script finishes seems like a non-issue to me.

Example migrate script:

```sql
create table if not exists customer
(
    email text
);

create unique index if not exists idx_custome_email on customer (email);
```

squawk output:

```
> squawk up.sql
up.sql:5:2: warning: require-concurrent-index-creation

   5 | create unique index if not exists idx_custome_email on customer (email);

  note: Creating an index blocks writes.
  help: Create the index CONCURRENTLY.

find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
```

I'd suggest relaxing the rule in cases where tables and their indexes are created in the same file.

**Comment by sbdchd** [comment]
Yeah if the table has just been created it won't matter since there's no data

**Comment by chdsbd** [comment]
If you wrap your migration in a transaction squawk won't worn:

```sql
begin;
create table if not exists customer
(
    email text
);

create unique index if not exists idx_custome_email on customer (email);
commit;
```

**Comment by hxeli** [comment]
wdyt about adding a rule that will ban concurrent index creations? We have a long list of migrations that runs on a new tenant and it may take a concurrent index creation forever to finish when running on a DB with intense throughput.

**Comment by sbdchd** [comment]
@hxeli does the wrapping in a database transaction not work in your case?

Couple other ideas:
- maybe we could check if the table was created in the same migration and assume it's okay?
- add a config option that lets you specify the behavior for the existing concurrent-index-creation rule, like `'never' | 'always'`
